ui/pages/confirmations/hooks/useEIP7702Account.ts

```diff
@@ -54,6 +54,32 @@ export function useEIP7702Account({
     [dispatch, globalNetworkClientId],
   );
 
+  const upgradeAccount = useCallback(
+    async (address: Hex, upgradeContractAddress: Hex) => {
+      const transactionMeta = (await dispatch(
+        addTransactionAndRouteToConfirmationPage(
+          {
+            authorizationList: [
+              {
+                address: upgradeContractAddress,
+              },
+            ],
+            from: address,
+            to: address,
+            type: TransactionEnvelopeType.setCode,
+          },
+          {
+            networkClientId: globalNetworkClientId,
+            type: TransactionType.batch,
+          },
+        ),
+      )) as unknown as TransactionMeta;
+
+      setTransactionId(transactionMeta?.id);
```

Comment:
Outside my scope of knowledge but would we want to call `setTransactionId` if that `id` is `undefined`?

Reply:
This fn call `await dispatch(addTransactionAndRouteToConfirmationPage(` generates `id`.